conf/spgwu.bess

```diff
@@ -169,9 +188,30 @@ class Port:
 
                 # Direct fast path traffic to Frag module
                 merge = __bess_module__("{}Merge".format(name), 'Merge')
-                frag = __bess_module__("{}IP4Frag".format(name), 'IPFrag')
-                merge -> frag:1 -> self.fpo
-                frag:0 -> Sink()
+                if ip_frag_flows is not None:
+                    frag = __bess_module__("{}IP4Frag".format(name), 'IPFrag')
+                    if MEASURE == 1:
+                        if self.nat is not None:
+                            merge -> 0:self.nat:1 -> frag:1 -> Measure(latency_ns_resolution=1, latency_ns_max=100000) -> self.fpo
+                        else:
+                            merge -> frag:1 -> Measure(latency_ns_resolution=1, latency_ns_max=100000) -> self.fpo
+                    else:
+                        if self.nat is not None:
+                            merge -> 0:self.nat:1 -> frag:1 -> self.fpo
+                        else:
+                            merge -> frag:1 -> self.fpo
+                    frag:0 -> Sink()
+                else:
+                    if MEASURE == 1:
+                        if self.nat is not None:
+                            merge -> 0:self.nat:1 -> Measure(latency_ns_resolution=1, latency_ns_max=100000) -> self.fpo
+                        else:
+                            merge -> Measure(latency_ns_resolution=1, latency_ns_max=100000) -> self.fpo
+                    else:
+                        if self.nat is not None:
+                            merge -> 0:self.nat:1 -> self.fpo
+                        else:
+                            merge -> self.fpo
```

Comment:
This is too branchy and repetitive. How best to simplify? Something like this pseudo code - 

```python
po = self.po
if Measure:
    m = Measure(latency_ns_resolution=1, latency_ns_max=100000)
    m -> self.fpo
    po = m

if ip_frag_flows:
    f = __bess_module__("{}IP4Frag".format(name), 'IPFrag')
    f:1 -> po
    po = f

if self.nat:
    self.nat:1 -> po
    po = self.nat

merge -> po
```

Reply:
same in other places

Reply:
should we add an ip checksum before going out? that catches all combinations of NAT and frag